js-wixcode-sdk/$w/repeater/guides/The Lifecycle of Repeated Items.md

```diff
@@ -0,0 +1,132 @@
+
+### Example:
+
+```javascript
+let data = [
+  { _id: "1", name: "Alice" },
+  { _id: "2", name: "Bob" },
+];
+
+$w.onReady(function () {
+  $w('#myRepeater').data = data;
+  $w('#myRepeater').onItemReady(($item, itemData) => {
+    $item('#nameText').text = itemData.name;
+  });
+
+  // Example: Add a new item when a button is clicked
+  $w('#addButton').onClick(() => {
+    // Create a new item with a unique _id
+    const newItem = { _id: "3", name: "John" };
+    data = [...data, newItem]; // Add the new item to the array
+    $w('#myRepeater').data = data; // Update the repeater with the new data
+  });
+   $w('#myRepeater').onItemReady(($item, itemData) => {
+    $item('#nameText').text = itemData.name;
+});
+});
+```
+
+
+## Update existing items
+
+When you update a repeater's `data` property, repeated items with the same _id values as before won't automatically display updated property values. Simply reassigning the `data` property doesn't refresh what's shown for those items.
+
+To ensure that any changes in your data are reflected in the UI, use the repeater's `forEachItem()` or `forItems()` functions. These functions update the elements inside each repeated item, so that the displayed content always matches the latest data in your array.
+### Example:
+
+```javascript
+const data = [
+  { _id: "1", name: "Alice" },
+  { _id: "2", name: "Bob" },
+  {_id: "3", name: "John"}
+];
+
+$w.onReady(function () {
+  $w('#myRepeater').data = data;
+  $w('#myRepeater').onItemReady(($item, itemData) => {
+    $item('#nameText').text = itemData.name;
+  });
+});
+
+data[0].name = "Alicia"; // Changed from Alice to Alicia
+
+  // Reassigning the data array alone will NOT update the repeated items:
+$w("#myRepeater").data = data; // No visible change
+
+  // To update the display, use forEachItem to manually refresh each element:
+$w("#myRepeater").forEachItem(($item, itemData, index) => {
+$item("#nameText").text = itemData.name; // This will now show "Alicia"
+});
+```
+
+## Remove items
+
+When you want to remove an item from a repeater, set the repeater's `data` property to a new array that no longer includes the object with the `_id ` of the item you want to remove.
+
+The repeater then checks which repeated items are represented in the new data array. For any repeated item whose `_id` is missing, the repeater removes the item.
+
+As each item is removed, the `onItemRemoved()` event handler runs and the display reflects the repeater without those items.
+
+```javascript
+const data = [
+  { _id: "1", name: "Alice" },
+  { _id: "2", name: "Bob" },
+  {_id: "3", name: "John"}
+];
+
+$w.onReady(function () {
+  $w('#myRepeater').data = data;
+  $w('#myRepeater').onItemReady(($item, itemData) => {
+    $item('#nameText').text = itemData.name;
+  });
+});
+// Remove the item with _id "2"
+data = data.filter(item => item._id !== "2");
+$w("#myRepeater").data = data; // The item for "Bob" is removed
+
```

Comment:
Again, let's do this in a button. And maybe do something more interesting like any item where the name is longer than 4 letters (if we stick with the name thing).